Update search command in DG (#108)
AY2122S2-CS2103T-T09-2/tp: docs/DeveloperGuide.md

_{more aspects and alternatives to be added}_

### Search applicant feature

> #### Rationale

Reviewer: Add empty line above for spacing

> `Search`  allows for a quick view of applicant's information in the linkedout book.

Reviewer 2: ```suggestion
`search` allows for a quick view of applicant's information in the application/ LinkedOUT.
```

> #### Implementation

Reviewer: Same, add an empty line above

> The proposed search mechanism is facilitated by `SearchCommandParser`. `SearchCommandParser` will map the creation of `KeywordsPredicate` based on the input prefix. `KeywordsPredicate` that supports the implementation:

Reviewer 2: ```suggestion
The proposed search mechanism is facilitated by `SearchCommandParser`. `SearchCommandParser` will map the creation of `KeywordsPredicate` based on the input prefix. `KeywordsPredicate` supports the following implementations:
```

* `NameContainsKeywordsPredicate` — Predicate which returns true if an applicant's full name matches partially with the input keyword.
* `JobContainsKeywordsPredicate` — Predicate which returns true if an applicant's job name matches partially with the input keyword.

> These predicates assist the filtering of applicant list in the `Model` interface, specifically for  `Model#updateFilteredApplicantList` and `Model#getFilteredApplicantList()`.

Reviewer 2: ```suggestion
These predicates assist the filtering of applicant list in the `Model` interface, specifically for  `Model#updateFilteredApplicantList()` and `Model#getFilteredApplicantList()`.
```

Given below is an example usage scenario and how the search mechanism behaves at each step.

Step 1. The user enters `search n/David` command to search for applicants in the linkedout book.

Step 2. The input keywords will be passed into `SearchCommandParser` and creates a `NameContainsKeywordsPredicate` if the keywords are not empty.

Step 3. The predicate is then passed into `Model#updateFilteredApplicantList` to filter and display applicants with partial name matching of "David" in the linkedout book.

> Step 4. The user enters `search j/Software Engineer` command to search for applicants in the linkedout book.

Reviewer: ```suggestion
Step 4. The user enters `search j/Software Engineer` command to search for applicants in the linkedout app.
```

Step 5. The input keywords will be passed into `SearchCommandParser` and creates a `JobContainsKeywordsPredicate` if the keywords are not empty.

> Step 6. The predicate is then passed into `Model#updateFilteredApplicantList` to filter and display applicants with partial job name matching of "Software" or "Engineer"  in the linkedout book.

Reviewer: ```suggestion
Step 6. The predicate is then passed into `Model#updateFilteredApplicantList` to filter and display applicants with partial job name matching of "Software" or "Engineer"  in the linkedout app.
```
